lib/record_store/provider/google_cloud_dns.rb

```diff
@@ -98,7 +98,8 @@ def build_from_api(record)
         when 'NS'
           record_params.merge!(nsdname: record.data[0])
         when 'SPF', 'TXT'
-          record_params.merge!(txtdata: record.data[0].gsub(';', '\;'))
+          txtdata = record.data[0].sub(/\A"(.*)"\z/, '\1').gsub('\"', '"').gsub(';', '\;')
+          record_params.merge!(txtdata: txtdata)
```

Comment:
Do you think it's worth moving the txtdata logic to the parent class and reuse it in the subclasses of Provider? I'm on the fence because the implementation is not quite the same, but looking at the regex, it feels like it would just work.